### PR TITLE
port prisma check -OCI VCN Security list has stateful security rules

### PR DESCRIPTION
**Pull request by JamesWoolfenden**
Addresses #1981

This check looks to see that security rules are stateful

### CODE REVIEW

File: checkov/terraform/checks/resource/oci/SecurityListIngressStateless.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            for rule in rules:
+                if 'stateless' in rule.keys():
+                    if rule.get("stateless") == [True]:
+                        return CheckResult.PASSED
+                    else:
+                        return CheckResult.FAILED
+            return CheckResult.PASSED
+
+        return CheckResult.UNKNOWN
+
```
Comment (rotemavni) [inline checkov/terraform/checks/resource/oci/SecurityListIngressStateless.py:25]: @JamesWoolfenden Please set the evaluated keys with the correct index, like we do here - https://github.com/bridgecrewio/checkov/blob/master/checkov/terraform/checks/resource/aws/RedShiftSSL.py

File: tests/terraform/checks/resource/oci/example_SecurityListIngressStateless/main.tf [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+resource "oci_core_security_list" "fail" {
+  compartment_id = oci_identity_compartment.tf-compartment.id
+  vcn_id         = oci_core_vcn.test_vcn.id
+      ingress_security_rules {
+        protocol = "all"
+        source="192.168.1.0/24"
+        stateless = false
+    }
```
Comment (rotemavni) [inline tests/terraform/checks/resource/oci/example_SecurityListIngressStateless/main.tf:27]: @JamesWoolfenden Can you please add a resource where `ingress_security_rules` doesn't exist?

**Review by rotemavni — COMMENTED** [review]
@JamesWoolfenden Great work! Please take a look at my comments 😄

File: checkov/terraform/checks/resource/oci/SecurityListIngressStateless.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+from checkov.common.models.enums import CheckCategories, CheckResult
+from checkov.terraform.checks.resource.base_resource_check import BaseResourceCheck
+
+
+class SecurityListIngressStateless(BaseResourceCheck):
+    def __init__(self):
+        name = "Ensure VCN inbound security lists are stateless"
+        id = "CKV_OCI_17"
+        supported_resources = ['oci_core_security_list']
+        categories = [CheckCategories.GENERAL_SECURITY]
```
Comment (gruebel) [inline checkov/terraform/checks/resource/oci/SecurityListIngressStateless.py:10]:
```suggestion
        categories = [CheckCategories.NETWORKING]
```
fits here better, also other ingress checks have that category.

File: checkov/terraform/checks/resource/oci/SecurityListIngressStateless.py (hunk 1 above)
Comment (JamesWoolfenden) [reply checkov/terraform/checks/resource/oci/SecurityListIngressStateless.py:25]: what do they do?

Comment (rotemavni) [reply checkov/terraform/checks/resource/oci/SecurityListIngressStateless.py:25]: The report will display the key where the violation occurred, and it also helps us create a fix for this check.

File: tests/terraform/checks/resource/oci/example_SecurityListIngressStateless/main.tf (hunk 2 above)
Comment (JamesWoolfenden) [reply tests/terraform/checks/resource/oci/example_SecurityListIngressStateless/main.tf:27]: added

**Review by gruebel — APPROVED** [review]
🚀

**Pull request merged**